controllers/azurejson_machine_controller.go

```diff
@@ -119,7 +118,7 @@ func (r *AzureJSONMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result,
 	}
 
 	// Fetch the Cluster.
-	cluster, err := util.GetOwnerCluster(ctx, r.Client, azureMachine.ObjectMeta)
+	cluster, err := util.GetClusterFromMetadata(ctx, r.Client, azureMachine.ObjectMeta)
```

Comment:
why switch here? I tested locally and seems to me that using owner is correct?

Reply:
tested locally == created a cluster and checked object meta for owners and labels. the azure machine template doesn't get a label corresponding to the cluster but does seem to get the owner ref. Not sure whether that itself is a problem.

Reply:
In my test; the AzureMachine doesn’t have the cluster as an owner, but has the label. It only has the Machine as the owner.

Reply:
nvm, this is for machine -_-

Reply:
yep -- was looking at template, sorry. would be good if we can get a test for this at some point, can do it separate from this PR though. Something like the reconciler tests we have elsewhere and then we can fetch the secret at the end and validate it exists -- kinda like https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/870/files#diff-af6683d08de01aaf9c4b27359a5d3c6c